File: OC/CoreAnimationTour/CoreAnimationTour/ViewController.m
//
//  ViewController.m
//  CoreAnimationTour
//
//  Created by 刘春奇 on 2017/1/11.
//  Copyright © 2017年 刘春奇. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.view.backgroundColor = [UIColor blackColor];
    UIView *view = [[UIView alloc]initWithFrame:CGRectMake(0, 0, 100, 100)];
    view.backgroundColor = [UIColor whiteColor];
    [self.view addSubview:view];
    view.center = self.view.center;
    
    //new a  layer
    CALayer *blueLayer = [[CALayer alloc]init];
    blueLayer.frame = CGRectMake(25, 25, 50, 50);
    blueLayer.backgroundColor = [UIColor blueColor].CGColor;
    [view.layer addSublayer:blueLayer];
    
//    blueLayer.contents
    
    
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: HCTreeTableView/HCTreeTableView/ViewController.swift
//
//  ViewController.swift
//  HCTreeTableView
//
//  Created by 刘春奇 on 2018/6/9.
//  Copyright © 2018年 Cloudnapps. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()


        let path = Bundle.main.path(forResource: "testJson", ofType: "json")
        
        //2 获取json文件里面的内容,NSData格式
        
        let jsonData=NSData(contentsOfFile: path!)
        
        //3 解析json内容
        
        let json = try? JSONSerialization.jsonObject(with: jsonData! as Data, options:[]) as! [[String:Any]]

        let treetableView  = TreeTableView.init(frame: self.view.bounds, style: .plain)
        self.view.addSubview(treetableView)
        treetableView.nodeData = json
        
        
        
        

    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

